addons/pos_razorpay/static/src/app/payment_razorpay.js

```diff
@@ -97,13 +135,37 @@ export class PaymentRazorpay extends PaymentInterface {
             "referenceId",
             referencePrefix + "/" + orderId + "/" + crypto.randomUUID().replaceAll("-", "")
         );
-        const data = {
-            amount: line.amount,
-            referenceId: localStorage.getItem("referenceId"),
-        };
-        return this._call_razorpay(data, "razorpay_make_payment_request").then((data) => {
-            return this._razorpay_handle_response(data);
-        });
+        if (order._isRefundOrder()) {
+            const data = {
+                amount: -line.amount,
+                externalRefNumber: localStorage.getItem("referenceId"),
+                transaction_id: line?.transaction_id,
+            };
+            const response = await this._check_payment_status(line);
+            if (response?.settlementStatus === "SETTLED") {
+                return this._call_razorpay(data, "razorpay_make_refund_request").then((data) => {
+                    return this._razorpay_handle_refund_response(data);
+                });
+            } else {
+                return this._call_razorpay(data, "razorpay_make_void_request").then((data) => {
+                    return this._razorpay_handle_refund_response(data);
+                });
+            }
```

Comment:
use the same method and pass refund_type as refund or void
```suggestion
            if (response?.settlementStatus === "SETTLED") {
                data.refund_type = 'refund'
            }
            else {
                data.refund_type = 'void'
            }   
            return this._call_razorpay(data, "razorpay_make_refund_request").then((data) => {
                    return this._razorpay_handle_refund_response(data);
            });
```